### ISSUE TITLE
is there a minimum android version?

### ISSUE THREAD

**Issue by krtschmr** [sa3dany/android-sms-hooks#5]

**Comment by sa3dany** [comment]
I guess the hard requirement is SDK version 30 (Android 11) which is required by [WorkManager](https://developer.android.com/jetpack/androidx/releases/work)

**Comment by krtschmr** [comment]
@sa3dany so i assume this phone https://www.amazon.de/gp/product/B09CKZ4HY5 won't work with this app? I'm asking since twilio isn't reliable anymore for me (the sender to blame, not twilio), so i'm in need to capture sms now from my sim card, rather than twilio and forward them to my server API to process the content. 
i thought i just buy a cheap android and get this running.
any recommendations on a smart phone? cheapest one is enough for me.
also: does this app run in background 24/7 or do i need to have it "open" all the time ( would put phone in the corner anyways, as it's sole purpose is to forward incoming SMS to my server)

**Comment by krtschmr** [comment]
https://www.amazon.de/-/en/Blackview-A70-Smartphone-without-Contract/dp/B08FX96PD this one would be android 11, i assume i can install the app then on there and have peace in mind? ✌🏿

**Comment by sa3dany** [comment]
Hi, Don't know why the [WorkManager page](https://developer.android.com/jetpack/androidx/releases/work) says that SDK version 30 is the minimum, I think it works reliably with SDK version 29 too (Android 10) according to the [code samples](https://github.com/googlecodelabs/android-workmanager/blob/master/build.gradle) used by Google.

**Comment by krtschmr** [comment]
i guess, price wise, it's either 65 EUR for android 8 phones or 100 EUR for android 11 phones. So i'm okay with buyin an 11 then. The most important thing is that all incoming sms will be forwarded reliable.

Does this app work 24/7 in the background? i read something that background apps are paused somehow by android, but since i'm a rails developer, i don't know much about phones

**Comment by sa3dany** [comment]
No, it used WorkManager to schedule tasks. The tasks run when there is an active network connection. There is currently no support for retries so if the network request made by the app failed for any reason (for example if the server returns a non-succes status code), that sms won't be retried again.

**Comment by krtschmr** [comment]
That's no problem, same for twilio. They won't resend. however i have 99.xx% uptime, so my server receives everything. if something slips through (once a year), i don't mind. 

I was more interested if this app is reliable on an android device. I guess i'll just order that phone as listed above and will try it out then on tuesday. #exciting.

**Comment by sa3dany** [comment]
You will also need to whitelist the app through any vendor specific battery optimization features to prevent the OS from freezing the app indefinitely until you reopen it.

As for reliability that's why I made the app after trying a number of other apps which would always stop sending the sms messages after some time.I haven't tried any other app since 2019 so I am not sure if the other available apps still have this issue.

The magic is the WorkManager which does the scheduling. So if you are worried about reliability you might look on how the work manager is implemented in the SDK version of the phone you will decide to buy.

**Comment by krtschmr** [comment]
This sounds like i should go the RaspberryPI route with a Sim module ;-)

I have ordered this phone now and if it doesn't work, i'll simply send it back. It's also okay for me to restart the phone every 2 weeks or so, if that's required. Since i haven't used android in forever, this is very new to me. Thank you for your input, appreciated.

**Comment by sa3dany** [comment]
No problem. Why would you need to restart the phone?

**Comment by krtschmr** [comment]
>  prevent the OS from freezing the app indefinitely until you reopen it.

I was irritated by this comment. I assume it meant it's okay for me to restart the app. 
I'll post a trip report by next weekend.  :-)

**Comment by sa3dany** [comment]
I actually meant the OS will freeze the app. This is usually okay for other app with have regular interaction since the OS just unfreezes the app, but since this is an set and forget kind of app with no further need to open the launcher activity, the OS battery saving feature will after some time freeze the app since it assumed the app is unused.

This is why you have to whitelist the app from battery saving.

**Comment by krtschmr** [comment]
Thanks for that info. It's the only app running on that phone. I'll then have a look for it and whitelist it accordingly.

Appreciated 🤝
